docs/source/usage/cleaning.rst

```diff
@@ -0,0 +1,189 @@
+Node Cleaning and Maintenance
+=============================
+
+Ironic provides two modes for node cleaning: automated and manual.
+
+Automated Node Cleaning
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Automated cleaning is automatically performed before the first workload has been assigned to a node and when hardware is recycled from one workload to another. Automatic cleaning can be customized for to enable or disable certain features of cleaning.
+
+Common use cases for automated cleaning:
+  * Erase disk drives and destroy all the metadata from previous user.
+  * Debug node failures resulting from disk failures by inspecting the automatic cleaning logs in Ironic conductor.
+  * Cut the time spent in manually wiping and cleaning the node after every workload, saving man hours.
+  * Reset raid configuration to a default state after every workload.
+
+Steps involved in automated cleaning:
+  1. Erase devices: out-of-band disk erase operation on all the supported physical drives in the node. Disk erase cannot be performed on physical drives.
+  2. Erase Devices Metadata: Wipe partition tables, signatures, file-system identifications, and other disk devices.
+  3. RAID Delete (Disabled-by-default): Delete any existing RAID configuration. This step needs to be enabled for automated cleaning.
+  4. Setup RAID configuration in advance(Disabled-by-default): Create RAID configuration from a predefined template when the node in enrolled. This step needs to be enabled for automated cleaning.
+
+
+Enable or Disable Automated Cleaning
+------------------------------------
+To enable automated cleaning, ensure that your ironic.conf has following option enabled.
+
+.. prompt:: bash $
+
+  [conductor] ironic.conf
+  automated_clean=true
+
+To disable it, ensure that your ironic.conf has following option disabled.
+
+.. prompt:: bash $
+
+  [conductor] ironic.conf
+  automated_clean=false
+
+Configuring Automated Cleaning Steps
+------------------------------------
+
+A cleaning step can be disabled by setting the priority for that cleaning step to zero or ‘None’. For instance, to disable erase_devices, you would set the following configuration option:
+
```

Comment:
Ah, you may have missed my suggestion about making a table of available automated cleaning steps, along with a short description and their default priority. The reason we want this is because it makes it immediately clear to someone reading the document what ESI will do by default; and what options are available.

Reply:
Was able to find some newly added automated cleaning steps from irmc and ilo driver pages, added them. Created two table, default steps and available steps